src/daft-micropartition/src/micropartition.rs

```diff
@@ -394,23 +394,22 @@ impl MicroPartition {
     }
 
     pub fn size_bytes(&self) -> DaftResult<usize> {
-        {
-            let guard = self.state.lock().unwrap();
-            if let TableState::Loaded(tables) = guard.deref() {
-                let total_size: usize = tables
-                    .iter()
-                    .map(|t| t.size_bytes())
-                    .collect::<DaftResult<Vec<_>>>()?
-                    .iter()
-                    .sum();
-                return Ok(total_size);
-            }
-        }
-        if let Some(stats) = &self.statistics {
+        let guard = self.state.lock().unwrap();
+        if let TableState::Loaded(tables) = guard.deref() {
+            let total_size: usize = tables
+                .iter()
+                .map(|t| t.size_bytes())
+                .collect::<DaftResult<Vec<_>>>()?
+                .iter()
+                .sum();
+            Ok(total_size)
+        } else if let Some(stats) = &self.statistics {
             let row_size = stats.estimate_row_size()?;
             Ok(row_size * self.len())
+        } else if let TableState::Unloaded(scan_task) = guard.deref() && let Some(size_bytes_file) = scan_task.size_bytes_file {
+            Ok(size_bytes_file as usize)
         } else {
-            // if the table is not loaded and we dont have stats, just return 0.
+            // if the table is not loaded, we dont have stats, and we don't have the file size in bytes, just return 0.
```

Comment:
Should we return `None` in that case as in... "I don't know"?

Reply:
We can change this to return a `DaftResult<Option<usize>>`, which should be fine for resource request usage, but I'll have to double-check other usages to make sure that a `None` won't blow things up.

Reply:
Turns out that we sum `size_bytes()` results in a few places in `physical_plan.py`, so we'd need to do this 0 fallback there anyways. How about we keep this 0 fallback for now?

Reply:
Sounds good to me, or we could also do None to 0 conversion in the physical
planner and add a loud comment there.

The case I’m worried about is if we don’t know the size of a file, and tell
the planner that it is 0, but the file in fact turns out to be huge 😝

At least if the planner knows the file size is unknown it can maybe handle
it a little differently in the future